Add JSpecify nullify to channel/interceptor, code, and codec/kryo packages

cppwfs opened the pull request:
* Missed the channel/interceptor on the last PR.
* Removed CompositeCodec(Codec defaultCodec)  because null can never be pass to the Codec.    In the past it would have thrown a NPE.

Continued effort or issue: https://github.com/spring-projects/spring-integration/issues/10083

cppwfs committed:
Add JSpecify nullify to channel/interceptor, code, and codec/kryo packages

* Missed the channel/interceptor on the last PR.
* Removed CompositeCodec(Codec defaultCodec)  because null can never be pass to the Codec.    In the past it would have thrown a NPE.

Continued effort or issue: https://github.com/spring-projects/spring-integration/issues/10083

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/codec/CompositeCodec.java:
```diff
@@ -35,18 +37,14 @@ public class CompositeCodec implements Codec {
-	public CompositeCodec(Codec defaultCodec) {
-		this(null, defaultCodec);
-	}
-
```
I agree.
Let's revert this ctor back and have another one with `@Nullable` for the `delegates` argument!

mrpiggi replied:
Alternatively, one could just use
```
	public CompositeCodec(Codec defaultCodec) {
		this(Map.of(), defaultCodec);
	}
```
and drop `@Nullable` for `this.delegates` as well as the null check in `findDelegate()`

cppwfs replied:
I agree with the  suggestion :
```
	public CompositeCodec(Codec defaultCodec) {
		this(Map.of(), defaultCodec);
	}
``` 
Else the user will just get a NPE downstream.

cppwfs committed:
Readd CompositeCodec ctor that calls super() without using null param

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/codec/CompositeCodec.java:
```diff
@@ -35,7 +37,7 @@ public class CompositeCodec implements Codec {
 	private final Codec defaultCodec;
 
-	private final Map<Class<?>, Codec> delegates;
+	private final @Nullable Map<Class<?>, Codec> delegates;
```
Then this should not be null

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/codec/CompositeCodec.java:
```diff
@@ -90,7 +92,7 @@ public <T> T decode(byte[] bytes, Class<T> type) throws IOException {
 
-	private Codec findDelegate(Class<?> type) {
+	private @Nullable Codec findDelegate(Class<?> type) {
 		if (this.delegates == null) {
```
And here we would use `isEmpty()` instead.

Everything else is good.

artembilan reviewed (changes requested):
Just one concern for the latest commit.

@mrpiggi ,

Do you have any other feedback, please?

cppwfs committed:
Update ctor to accept a null map of delegates

cppwfs committed:
Delegates should not be nullable in CompositeCodec

Removed @Nullable from delegates field in CompositeCodec
Remove check for null for delegate and replace it with isEmpty

cppwfs committed:
Remove @Nullable from delegates param in the ctor

mrpiggi commented:
@artembilan So you obviously voted against my suggestion regarding `findDelegate(Class<?> type)` to never return `null` from this method but always fall back to `this.defaultCodec` and get rid of all `null` checks after calling this method. Just an oversight or is there any< reason?

artembilan commented:
>  you obviously voted against my suggestion

No, I did not voted against you.
Sorry about that: was concentrated to make those nullables OK.

Feel free to raise a PR with whatever improvements you see!
Thank you for your help!
